src/server/socksSocket.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import net from 'net';
+import { EventEmitter } from 'events';
+
+import { SdkObject } from './instrumentation';
+import { debugLogger } from '../utils/debugLogger';
+import { isLocalIpAddress } from '../utils/utils';
+import { SocksProxyServer, SocksConnectionInfo, SocksInterceptedHandler } from './socksServer';
+import { LaunchOptions } from './types';
+
+export class BrowserServerPortForwardingServer extends EventEmitter {
```

Comment:
I'd inline this file into browserServerImpl.ts, and merge SocksSocket with SocksSocketDispatcher right away.

Reply:
I have the assumption that this get's also in the future used in playwrightclient, so I would keep it separate for now.